partiql-lang/src/main/kotlin/org/partiql/lang/CompilerPipeline.kt

```diff
@@ -47,7 +47,7 @@ data class StepContext(
      * Includes built-in functions as well as custom functions added while the [CompilerPipeline]
      * was being built.
      */
-    val functions: @JvmSuppressWildcards Map<String, ExprFunction>,
```

Comment:
We need to deprecate this constructor and offer another constructor that allows for a list.

Reply:
I don't think we need to deprecate this because the data class StepContext is only invoked inside CompilerPipelineImpl where we already add deprecate notice to the old functionMap. In this way, CompilerPipelineImpl always convert functionMap to a list so StepContext will just use list.